custom_components/resmed_myair/client/rest_client.py

```diff
@@ -0,0 +1,704 @@
+import base64
+import datetime
+import hashlib
+from http.cookies import SimpleCookie
+import logging
+import os
+import re
+from typing import Any, List
+from urllib.parse import DefragResult, parse_qs, urldefrag
+
+from aiohttp import ClientResponse, ClientSession
+from aiohttp.http_exceptions import HttpProcessingError
+from homeassistant.helpers.redact import async_redact_data
```

Comment:
It would be best to keep HA dependencies out of this file so it can be made into a separate librar(like #45 ). though looking at the code for this function it is pretty simple and it could be copied then, but we should be careful with any other dependencies.

Reply:
Understood. I'd actually like to redo this as part of the logger and not have to call it explicitly. When I do that, I'll make it a local method and not include anything from HA. If you're ok with it, I'll likely do that in a future PR.